accepted/0000-staged-packages.md

```diff
@@ -0,0 +1,90 @@
+# RFC: Staged Packages
+
+## Summary
+
+Staging workflow which allows CI to publish new package versions, while allowing
+humans to be the final arbiters of general availability.
+
+## Motivation
+
+Anything published by CI is instantly available for general use, which isn't
+ideal when a human is required to be in the loop. And even when fully automated
+CI/CD is allowed, the inability to use 2FA with auth tokens is a security risk.
+
+## Detailed Explanation
+
+We need a minimum of four things to solve these problems:
+
+1. Staging area on the registry invisible to most `npm` commands
+2. Auth tokens that can be scoped to the staging area
+3. CLI flag which lets `npm` commands act on the staging area
+4. CLI command that lets users promote a staged package
+
+## Rationale and Alternatives
+
+One alternative is to defer to third-party repository managers for staging
+functionality. However, this requires that users set up and manage yet another
+tool, which most won't bother to do. In addition, even if users do acquire those
+tools, because staging commands aren't built into the npm CLI, they have to
+learn a new interface to perform staging operations, which adds unnecessary
+friction to the developer experience and again encourages the wrong sorts of
+publishing behaviors.
+
+Another alternative is to build this using our existing `dist-tag`
+functionality. The appeal is that it uses a mechanism we've already built. The
+downside is that we would have to actively ignore these tagged packages in the
+CLI and would be eating up versions as well, not ideal.
+
+Yet another alternative is to build a fancy release manager application,
+separate from the CLI. However, this is a larger project, and we can almost
+certainly implement a useful staging workflow based on tooling that already
+exists. Also, even if we start by adding this to the CLI, it doesn't mean we
+can't build a more complex application on top of the CLI later on.
+
+After considering these alternatives, we belive a staging workflow built into
+the npm CLI provides the shortest path to useful staging functionality that we
+can iterate upon. It will allow us to test these new concepts in a tool with
+which we and our community are both familiar.
+
+## Implementation
+
+* Add a `stagedVersions` object to the packument which lists all packages that
+  have been staged and not promoted.
+* A new `--stage` flag for the `npm publish` command that allows a package to be
+  published to staging.
+* A new `--stage` flag for the `npm install` command that allows a package to be
```

Comment:
It might be worth clarifying here that semver rules still apply (i.e. the staged version, if any, will only be installed if it matches the specifier in `package.json`)

Reply:
Now that I am reading this, I feel like one other addition should be made here.  Install `--stage` would only work with a single package and a specific version. Example:

Good:
```
$ npm i --stage foo@1.0.0
```

Bad:
```
$ npm i --stage
$ npm i --stage foo
$ npm i --stage foo@^1.0.0
$ npm i --stage foo@latest
```

Again with the goal that installing from the stage be hard and a very explicit action.